lib/pingback.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Pingback one url
+ */
+
+'use strict';
+
+const URL = require('url');
+const http = require('http');
+const https = require('https');
+const querystring = require('querystring');
+
+module.exports = (config) => {
+	const cfg = config;
+	const url = cfg.url;
+	const parsedUrl = URL.parse(url);
+	const client = parsedUrl.protocol === 'https:' ? https : http;
+
+	if (!parsedUrl.hostname) {
+		return {
+			ok: false,
+			error: 'No hostname found for ' + url,
+			send: () => {}
+		};
+	}
+
+	const send = (data, cb) => {
+		if (data && data.result && data.action && data.url) {
+			const postData =  querystring.stringify({
+				result: data.result,
+				action: data.action,
+				url: data.url,
+				source: 'algolia-crawler'
+			});
+
+			const httpOptions = {
+				hostname: parsedUrl.hostname,
+				port: parsedUrl.port || (parsedUrl.protocol === 'https:' ? 443 : 80),
+				path: parsedUrl.path || '/',
+				method: 'POST',
+				auth: config.http && config.http.auth,
+				headers: {
+					'Content-Type': 'application/x-www-form-urlencoded',
+					'Content-Length': postData.length
+				}
+			};
+	
+			var req = client.request(httpOptions, (res) => {
+				let data = '';
+				res.setEncoding('utf8');
+
+				//Manage error
+				if (res.statusCode !== 200) {
+					cb({
+						ok: false,
+						message: 'HTTP error ' + res.statusCode + ' ' + url
+					});
+					return;
+				}
+
+				//Grab all data
+				res.on('data', (chunk) => {
+					data += chunk;
+				});
+
+				//When done.
+				res.on('end', (chunk, encoding) => {
+					if (!!chunk) {
+						data += chunk;
+					}
+
+					cb({
+						ok: true
+					});
+				});
+			});
+			
+			const callback = (err) => {
+				cb({
+					ok: false,
+					err: err
+				});
+			};
+			req.on('error', (e) => callback(e));
```

Comment:
There are too many functions here. Either pass callback directly to error, or inline it. I think that the inline is the best.